skdownscale/pointwise_models/gard.py

```diff
@@ -121,11 +149,32 @@ def _predict_one_step(self, lr_model, X):
         x = np.asarray(self.kdtree_.data)[inds]
         y = self.y_[inds]
 
-        # train linear regression model
-        lr_model.fit(x, y)
+        # figure out if there's a threshold of interest
+        if self.thresh is not None:
+            exceed_ind = y > self.thresh
+        else:
+            exceed_ind = np.array([True] * len(y))
+
+        # train logistic regression model
+        binary_y = exceed_ind.astype(int)
+        if not binary_y.mean() == 1:
+            logistic_model.fit(x, binary_y)
+            exceedance_prob = logistic_model.predict_proba(X)[:, 0][0]
+        else:
+            exceedance_prob = 1.0
+
+        self.exceedance_prob_[i] = exceedance_prob
```

Comment:
I think this can be cleaned up a bit. Thoughts on this implementation:

```suggestion
        binary_y = exceed_ind.astype(np.int8)
        if not np.all(binary_y == 1):
            logistic_model.fit(x, binary_y)
            self.exceedance_prob_[i] = logistic_model.predict_proba(X)[:, 0][0]
        else:
            self.exceedance_prob_[i] = 1.0
```

And question:

Is `logistic_model.predict_proba(X)[:, 0][0]` different than `logistic_model.predict_proba(X)[0, 0]`?

Reply:
sounds good and they should be the same, updated